### DISCUSSION TITLE
Thread 1: "Index 1 is out of bounds (must be less than 1)."

### DISCUSSION THREAD

**Discussion by atsujr** [lit-iphone-course/lit-iphone-course#20] [Question-original] [answered]
(1)実現したいこと
tableviewに打ち込んだ内容をrealmを用いて追加したい。
saveボタンを押し、元の画面に戻ろうとした時にアプリが落ちてしまう。その後もう一度アプリを実行すると、最初の画面に先ほど打ち込んだ内容が保存されている。
(2)(3)調べたこと、実装したこと
Thread 1: "Index 1 is out of bounds (must be less than 1)."
というエラーを調べたところ、Realmに保存されているモデルの件数と表示されているTableViewの件数(TableViewの認識している件数が以前のままで)の整合性がとれなくエラーになるということがわかり、
https://qiita.com/masa283/items/5cec609fa73df3c3c2c7
を参考に、Listをextensionしてみたが、やはり打ち込んで、saveボタンを押し、元の画面に戻ろうとした時にアプリが落ちてしまう。
(4)ここではないかというところ
先ほど調べた内容的に、  
func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return realm.objects(TodoList.self).count
        
    }
ここのような気がしています。ご確認、よろしくお願いします。gitのURLは以下のものです。
https://github.com/atsujr/todo_day2

**Comment by appleEmily** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/49332380/189339362-6d51494e-a183-4f20-bf5b-1fb00b5e3e6d.png)

viewDidLoadのところでrealmのデータを取ってきてると思うけど、画面が戻った時を考えるとviewWillAppearでやらないとじゃないかしら？

**Reply by atsujr** [reply]
override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        tableview.reloadData()
        
            todolist = realm.objects(TodoList.self).first?.list
            tableview.reloadData()

    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        tableview.delegate = self
        tableview.dataSource = self
        
        tableview.reloadData()
    }
これでもう一回実行してみたんですけど、今度は1個目はsaveできて、tableviewに表示できるんですけど、2個目をsaveして、tableviewの画面に戻るとまた同じエラーが出てアプリがクラッシュしちゃいます、、
